src/OrleansSQLUtils/Storage/Provider/AdoGrainKey.cs

```diff
@@ -73,17 +74,18 @@ public byte[] GetHashBytes()
                 bytes = ToGuidKey(N0Key, N1Key).ToByteArray();
             }
 
+            var utf8 = new UTF8Encoding();
```

Comment:
@veikkoeeva can we do a repro for the scenario which produces the invalid output? I'm not convinced that this is needed, since the encoding classes are stateless in general.

@sergeybykov  I'd not go the ThreadLocal way since I thing that's not available in .Net Core, I just don't see the real need for this, were you bitten by it before?